src/objects/impls/structure_tower.rs

```diff
@@ -1,18 +1,27 @@
 use crate::{
     constants::ReturnCode,
-    objects::{Creep, Structure, StructureTower},
+    objects::{Attackable, SharedCreepProperties, StructureProperties, StructureTower},
 };
 
 impl StructureTower {
-    pub fn attack(&self, target: &Creep) -> ReturnCode {
+    pub fn attack<T>(&self, target: &T) -> ReturnCode
+    where
+        T: Attackable,
+    {
         js_unwrap! { @{self.as_ref()}.attack( @{target.as_ref()} ) }
     }
 
-    pub fn heal(&self, target: &Creep) -> ReturnCode {
+    pub fn heal<T>(&self, target: &T) -> ReturnCode
```

Comment:
Can you heal enemy creeps?

Reply:
Yup, sure can!  You usually wouldn't want to, but there is a useful uncommon case in healing allies' creeps (though my intent for this change was just to make it consistent with the args for the heal function on `Creep`)